partcad/src/partcad/actions/part_actions.py

```
@@ -46,42 +67,44 @@ def import_part_action(project: Project, kind: str, name: str, source_path: str,
+    target_folder.mkdir(parents=True, exist_ok=True)
+    target_path = (target_folder / f"{name}.{kind}").resolve()
 
+    # Log the target path and copy if needed
     if target_path.exists() and source_path.samefile(target_path):
-        pc_logging.warning(f"Skipping copy: source and target paths are the same ({source_path}).")
+        pc_logging.warning(f"Skipping copy: source and target are the same ({source_path}).")
     else:
         try:
             shutil.copy2(source_path, target_path)
         except shutil.Error as e:
             raise ValueError(f"Failed to copy '{source_path}' -> '{target_path}': {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)